controllers/storage_test.go

```diff
@@ -218,6 +240,19 @@ func TestStorage_Archive(t *testing.T) {
 			},
 			wantErr: false,
 		},
+		{
```

Comment:
Can you please add the following to the "exclude VCS" test:

```go
			},
			wantDirs: []string{"!.git"},
```

(this will now make the tests fail).

To solve this (and prevent including e.g. empty `.git/` directories), I think some of the patterns in https://github.com/fluxcd/source-controller/blob/main/pkg/sourceignore/sourceignore.go#L31 and https://github.com/fluxcd/source-controller/blob/main/pkg/sourceignore/sourceignore.go#L33 need revising.